pkg/controllers/management/auth/auth_config_test.go

```diff
@@ -91,34 +94,34 @@ func TestCleanupRuns(t *testing.T) {
 		},
 	}
 
-	mockAuthConfig := newMockAuthConfigClient()
-
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			mockUsers := newMockUserLister()
-			var service cleanupService
-			controller := authConfigController{
-				cleanup:          &service,
-				authConfigClient: mockAuthConfig,
-				users:            &mockUsers,
-			}
 			config := &v3.AuthConfig{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:        azuread.Name,
 					Annotations: map[string]string{CleanupAnnotation: test.annotationValue},
 				},
 				Enabled: test.configEnabled,
 			}
+			var service cleanupService
+			controller := authConfigController{
+				cleanup:                 &service,
+				authConfigsUnstructured: newMockAuthConfigClient(config),
+				users:                   &mockUsers,
+			}
 
-			obj, err := controller.sync("test", config)
+			authConfig, err := controller.sync("test", config)
+			acObject := authConfig.(*v3.AuthConfig)
 			require.NoError(t, err)
+			assert.Equal(t, test.newAnnotationValue, acObject.Annotations[CleanupAnnotation])
 			assert.Equal(t, test.expectCleanup, service.cleanupCalled)
-			assert.Equal(t, test.newAnnotationValue, obj.(*v3.AuthConfig).Annotations[CleanupAnnotation])
 		})
 	}
 }
 
 func TestAuthConfigSync(t *testing.T) {
+	t.Parallel()
```

Comment:
Can you explain a bit more why you moved this up here?

Reply:
This is something that has bitten me before but not consistently. Basically, a test itself should (if possible) be run in parallel with other tests. We can also run subtests in parallel, but then we should also ensure there won't be that variable shadowing problem, where only the last subtest will be run due to scoping and goroutine special case in Go.
I don't know if this always happens in t.Run subsets. I know in regular for loop local variables and goroutine spawns, this happens every time.
I noticed that there is this `t.Parallel` without copied variables line in your 2.6 part of the test, I ran with and without the line, and the tests passed, but I can't prove that with it, all the subtests actually run.
Maybe @KevinJoiner knows how this works in subtests (if it's safe to just call `t.Parallel` without loop variables copied).

Reply:
L162 which provides the `test := test` is what makes it work. If you'd like to confirm that this test is running as expected, you can add a logrus.Errrof() line at the start of each test that says something like `running test %s, test.name`.

I don't think we should be changing this without a good reason. If you can demonstrate that the test fails occasionally, or isn't running, then I'm ok with changing this. Otherwise, please leave it as is.

Reply:
You're right, my fault, missed this. Also, it's my mistake in v2.6 - it's my test that misses that `test := test`:
https://github.com/rancher/rancher/blob/6d1bd94edf59b130caf51e766488cef15875eac3/pkg/controllers/management/auth/auth_config_test.go#L101